src/modules/.internal/polyfills/intlFormatToPartsPolyfill.ts

```diff
@@ -0,0 +1,65 @@
+declare global {
+  // Extend the NumberFormat interface to include a custom method formatToParts
+  interface NumberFormat {
+    formatToParts?(number: number): Intl.NumberFormatPart[];
+  }
+}
+
+// Define a custom structure to represent parts of a formatted number
+type CustomNumberFormatPart = {
+  type: 'string' | 'integer'; // Define a custom type matching the options
+  value: string;
+};
+
+// Create a custom enum-like type for the part types
+type CustomNumberFormatPartType = 'string' | 'integer';
+
+// Function to map custom type to Intl.NumberFormatPart
+function mapToNumberFormatPart(
+  parts: CustomNumberFormatPart[],
+): Intl.NumberFormatPart[] {
+  // Map the custom number format parts to the standard Intl.NumberFormatPart
+  return parts.map((part) => ({
+    type: part.type as Intl.NumberFormatPartTypes, // Casting the custom type to the standard type
+    value: part.value,
+  }));
+}
+
+// Export a default function for the Intl polyfill
+export default function intlFormatToPartsPolyfill() {
+  // Check if the formatToParts method is not defined
+  if (!Intl.NumberFormat.prototype.formatToParts) {
```

Comment:
As far as I can see, `intlFormatToPartsPolyfill` will be triggered even if `Intl` is `undefined`. Wouldn't this throw an error in the condition here ?